src/compute/src/compute_state.rs

```diff
@@ -169,6 +179,69 @@ impl SinkToken {
     }
 }
 
+async fn persist_peek(
```

Comment:
Could we extract all logic for `persist_peek` into a different file/module? That would help enforcing structure and making it clearer that there are two distinct types of peeks.

Reply:
I hesitate because:
- It's only a hundred lines or so of persist-specific logic.
- I don't see a good reason to shift out one peek variant but not the other.

But I certainly will extract if y'all prefer it! Might do this last to avoid churn.